src/Microsoft.AspNetCore.SignalR.Core/HubEndPoint.cs

```diff
@@ -476,6 +477,16 @@ private bool IsChannel(Type type, out Type payloadType)
             }
             catch (Exception ex)
             {
+                if (ex is OperationCanceledException)
+                {
+                    // If a streaming method was canceled we don't want to send a HubException message - this is not an error case
+                    if (connection.ActiveRequestCancellationSources.TryGetValue(invocationId, out var cts) && cts.IsCancellationRequested)
```

Comment:
It would be cleaner to just store an error message object and have a single code path that creates the StreamingCompletion.